transport/http/handler.go

```diff
@@ -265,6 +271,29 @@ func (rw *responseWriter) SetApplicationError() {
 	rw.w.Header().Set(ApplicationStatusHeader, ApplicationErrorStatus)
 }
 
+func (rw *responseWriter) SetApplicationErrorMeta(meta *transport.ApplicationErrorMeta) {
+	if meta == nil {
+		return
+	}
+	if meta.Code != nil {
+		rw.w.Header().Set(_applicationErrorCodeHeader, strconv.Itoa(int(*meta.Code)))
+	}
+	if meta.Name != "" {
+		rw.w.Header().Set(_applicationErrorNameHeader, meta.Name)
+	}
+	if meta.Message != "" {
+		rw.w.Header().Set(_applicationErrorMessageHeader, truncateAppErrMessage(meta.Message))
+	}
+}
+
+func truncateAppErrMessage(val string) string {
+	if len(val) <= _maxAppErrMessageHeaderLen {
+		return val
+	}
+	stripIndex := len(val) - len(_truncatedHeaderMessage) - 1
```

Comment:
Shouldn't `stripIndex := maxAppErrMessageHeaderLen - len(_truncatedHeaderMessage)`?
That way the resultant string with `_truncatedHeaderMessage` appended would always == `maxAppErrMessageHeaderLen`

Reply:
nice catch, looks like I didn't write an extensive test! looks like I made this same mistake in the TChannel impl, I'll add that in another commit.